### PR TITLE
Chore/apply eslint and prettier

### PR DESCRIPTION
**Pull request by dawntancarrasco**
- added eslint and prettier
- added script in package json
- changed overflow scroll of modal to auto
- formatted code when running prettier
- resolved linting issues

Screenshot 2023-07-31 at 1 58 05 PM
Screenshot 2023-07-31 at 1 58 16 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,7 +78,7 @@
     "storybook-addon-designs": "^6.3.1",
     "style-dictionary": "^3.7.2",
     "styled-components": "5.3.8",
-    "typescript": "^4.9.5",
+    "typescript": "*",
```
Comment (sbracegirdle) [inline package.json:81]: Minor issue; depending on`*` is not generally a good idea and could cause issues on npm install / upgrade

File: package.json [hunk 2]
```diff
@@ -48,13 +50,24 @@
...
     "@types/react-helmet": "^6.1.6",
     "@types/styled-components": "^5.1.26",
+    "@typescript-eslint/eslint-plugin": "^6.2.0",
+    "@typescript-eslint/parser": "^6.2.0",
     "babel-jest": "^29.5.0",
     "babel-loader": "^9.1.2",
     "csstype": "^3.1.1",
+    "eslint": "^8.45.0",
+    "eslint-config-standard-with-typescript": "^37.0.0",
+    "eslint-plugin-import": "^2.25.2",
```
Comment (sbracegirdle) [inline package.json:60]: Question; so what plugins have we added here, what does each give us?

What other plugins have been considered? For example. did you look at any of the following?

- eslint-plugin-react-hooks
- eslint-plugin-jsx-a11y
- eslint-plugin-styled-components-a11y
- eslint-plugin-prettier
- eslint-plugin-storybook
- eslint-plugin-testing-library
- eslint-plugin-jest
- eslint-plugin-unicorn
- eslint-plugin-security
- eslint-plugin-react-perf

File: package.json [hunk 3]
```diff
@@ -9,6 +9,8 @@
     "prestart": "yarn run build:tokens",
     "start": "NODE_OPTIONS=--openssl-legacy-provider start-storybook -p 6006",
     "build": "NODE_OPTIONS=--openssl-legacy-provider build-storybook",
+    "format": "prettier --write \"src/**/*.{ts,tsx}\"",
+    "lint": "eslint --max-warnings=0 .",
```
Comment (sbracegirdle) [inline package.json:13]: Suggestion; can we eslint and prettier checks to the CI as part of this please?

File: package.json [hunk 4]
```diff
@@ -9,6 +9,8 @@
     "prestart": "yarn run build:tokens",
     "start": "NODE_OPTIONS=--openssl-legacy-provider start-storybook -p 6006",
     "build": "NODE_OPTIONS=--openssl-legacy-provider build-storybook",
+    "format": "prettier --write \"src/**/*.{ts,tsx}\"",
```
Comment (sbracegirdle) [inline package.json:12]: Suggestion; can we adopt the prettierrc file from webapp for consistency across GovConnex projects please?

File: package.json (hunk 2 above)
Comment (dawntancarrasco) [reply package.json:60]: "eslint-config-standard-with-typescript": "^37.0.0"
This package provides an ESLint configuration that includes rules for the Standard style guide along with support for TypeScript-specific rules. It ensures that your TypeScript code adheres to the Standard style guide conventions.

"eslint-plugin-import": "^2.25.2"
This plugin provides ESLint rules for linting and enforcing best practices related to import statements in your code. It helps ensure consistent and correct import usage, such as detecting unused imports, enforcing specific import order, and preventing circular dependencies.

"eslint-plugin-n": "^15.0.0 || ^16.0.0"
This plugin helps you enforce rules based on the Node.js version you are targeting. It allows you to define rules that are specific to a particular Node.js version or range. For example, you might have different APIs available in different Node.js versions, and this plugin helps you handle those variations.

"eslint-plugin-promise": "^6.0.0"
This plugin provides ESLint rules for enforcing best practices and conventions related to JavaScript promises. Promises are used for handling asynchronous operations, and this plugin helps catch potential issues and enforce consistent usage.

"eslint-plugin-react": "^7.33.0"
This plugin provides ESLint rules specifically designed for React applications. It helps ensure best practices, detect potential issues, and enforce coding standards related to React components and JSX syntax.

"eslint-plugin-storybook": "^0.6.13"
This plugin is intended for projects that use Storybook, a popular tool for developing UI components in isolation. The plugin provides ESLint rules to ensure consistency and best practices when writing stories and configuring Storybook.

As discussed, will take note of the other plugins as well.

File: package.json (hunk 4 above)
Comment (dawntancarrasco) [reply package.json:12]: yup i have added this :)

**Comment by sbracegirdle** [comment]
Looks good! Btw the check failed, it found a code formatting issue!

```
[warn] src/components/Button/Button.tsx
[warn] Code style issues found in the above file. Run Prettier to fix.
```

**Review by sbracegirdle — APPROVED** [review]

**Pull request merged**